source/server/overload_manager_impl.cc

```diff
@@ -75,22 +75,29 @@ class ScaledTriggerImpl final : public OverloadAction::Trigger {
  */
 class ThreadLocalOverloadStateImpl : public ThreadLocalOverloadState {
 public:
+  ThreadLocalOverloadStateImpl(const NamedOverloadActionSymbolTable& action_symbol_table)
+      : action_symbol_table_(action_symbol_table),
+        actions_(action_symbol_table.size(), OverloadActionState(0)) {}
+
   const OverloadActionState& getState(const std::string& action) override {
-    auto it = actions_.find(action);
-    if (it == actions_.end()) {
-      it = actions_.insert(std::make_pair(action, OverloadActionState::inactive())).first;
+    if (auto symbol = action_symbol_table_.find(action); symbol != absl::nullopt) {
+      return actions_[symbol->index()];
     }
-    return it->second;
+    return always_inactive_;
   }
 
-  void setState(const std::string& action, OverloadActionState state) {
-    actions_.insert_or_assign(action, state);
+  void setState(NamedOverloadActionSymbolTable::Symbol action, OverloadActionState state) {
+    actions_[action.index()] = state;
```

Comment:
Is there some way to assert at ctor time that `action_symbol_table.size()` is >= the largest possible value of any `action.index()`? Maybe I am misunderstanding how `.index()` works but it seems like it would be easy to later accidentally modify this code in a way that causes failed vector lookups.